### ISSUE TITLE
Wrong cross-selling and preview thumbnails are loaded on product detail pages / wrong value for the sizes attribute of these thumbnails

### ISSUE THREAD

**Issue by gamerlion7** [FriendsOfShopware/FroshLazySizes#34]
### Actual behaviour

Images from cross-selling products are 400 x 400 pixels big although there exist thumbnails with 200 x 200 and 280 x 280 pixels. The sizes attribute of cross-selling products has a value of 274 pixels.
The small preview images are 200 x 200 pixels big although there exist thumbnails with 56 x 56 and 75 x 75 pixels. The sizes attribute of the small preview images has a value of 75 pixels.

### Expected behaviour

Images from cross-selling products are 200 x 200 pixels big, if there are thumbnails of this size. The sizes attribute of these images has a value of 200 pixels.
The small preview images are 56 x 56 pixels big, if there are thumbnails of this size. The sizes attribute of these images has a value of 56 pixels.

### Steps to Reproduce?

Load and inspect a product detail page.

**Comment by tinect** [comment]
At the moment I am away from keyboard.

274px is the effective possible width, isn‘t it?

Do the products have a set aspectratio in code?  did you check the special plugin config. Maybe you can work with this.

**Comment by tinect** [comment]
I've had a chance to have a look, and you're right—some styles do seem to be missing in the storefront.

Which version of Shopware are you using? I couldn't replicate the cross-selling issue on version 6.6, but I did notice it on a 6.4 shop. As for the preview/thumbnail images, I was able to reproduce that problem.

You are welcome to dig deeper into the issue, and if you'd like, you can submit a pull request this repo or the [library](https://github.com/aFarkas/lazysizes) we're using.

**Comment by gamerlion7** [comment]
I noticed it in Shopware 6.4.20.2.

The cross-selling thumbnails have a max-height of 200 pixels by default. If it is a square image, the size attribute should have a value of 200 pixels, shouldn't it?

**Comment by tinect** [comment]
> The cross-selling thumbnails have a max-height of 200 pixels by default. If it is a square image, the size attribute should have a value of 200 pixels, shouldn't it?

You are right. This is a generic style and behaviour problem in the storefront. The library cannot detect the correct size initially. You can verify it by resizing the browser afterwards and see there will be the correct size of 200px.
